internal/pubsub/pubsub.go

```diff
@@ -237,6 +249,14 @@ func (s *Subscription) As(i interface{}) bool {
 	return s.driver.As(i)
 }
 
+// MessageAs converts a message to an implementation-specifc message
+// type, returning true or false depending on whether the conversion
+// succeeded.
+func (s *Subscription) MessageAs(m *Message, i interface{}) bool {
+	dm := toDriverMessage(m)
```

Comment:
Hmm, at this point we've already lost information... not sure how to do this. I guess `pubsub.Message` should hold a `driver.Message` which should hold the raw driver message type.